src/run/run.c

```diff
@@ -75,6 +76,20 @@ STATIC_DESTRUCTOR_REGISTER(arg_timer_property, strv_freep);
 STATIC_DESTRUCTOR_REGISTER(arg_working_directory, freep);
 STATIC_DESTRUCTOR_REGISTER(arg_cmdline, strv_freep);
 
+static void suicidep(int *signal) {
+        /* Kill ourselves with a specified signal. This propogates the signal that killed our child process
+         * to the caller, allowing for example the shell to properly report what happened to the child.
+         *
+         * We do as a static destructor to do it as close as possible to the process exit.
+         * This should be the last static destructor.
+         */
+
+        if (*signal > 0)
+                kill(0, *signal);
```

Comment:
hmm, uh, this is problematic, not all signals kill processes... moreover we block at least SIGWINCH, SIGTERM, SIGINT in some cases in run.c, and raising a blocked signal doesn't make too much sense?

Also the libc call for killing ourselves is `raise()`... ;-)

I think we should only do this for some signals here if at all, and if we do this, then we should unmask the signal first, and reset the signal handler?

AFAIK shells generally do not propagate signals as signals, but just as exit code 255 or 127, maybe we should do the same and not bother with propagating correct exit by signal?